### ISSUE TITLE
Setup Buildroot Repository

### ISSUE THREAD

**Issue by geoffreyjensen6** [cu-ecen-aeld/final-project-geoffreyjensen6#1]
**Task:** Create a Buildroot repository for the project, targeting Beaglebone Black Hardware

**DoD:** Build is successful, able to program image to Beaglebone Black and boot.

**Blockers:** None

**Status:** This issue is complete as of 5/22/2023 with commit a1487fd. The main difficulty in moving a latest kernel & uboot was really the uboot version. Able to boot Beaglebone Black using my buildroot configuration with uboot 2023.04 and kernel 5.4.70.

![Image](https://github.com/cu-ecen-aeld/final-project-geoffreyjensen6/assets/123145400/9ba9cb2e-d780-42d4-80fd-44437253efcc)

![Image](https://github.com/cu-ecen-aeld/final-project-geoffreyjensen6/assets/123145400/d7ff20ff-bd6a-49dd-96e2-f010271821b9)

**Comment by geoffreyjensen6** [comment]
Took a few tries initially but following details from https://github.com/cu-ecen-aeld/buildroot-assignments-base/wiki/Beaglebone-Black-Hardware-Support and using the example config file on the provided working branch as a reference https://github.com/cu-ecen-aeld/buildroot-assignments-base/blob/hw-beaglebone-black-br-2019.05/base_external/configs/aesd_beaglebone_defconfig i was able to successfully boot Linux on Beaglebone Black. I am going to continue working to upgrade kernel version and uboot version to newer revs though because this defconfig references older versions.

**Comment by geoffreyjensen6** [comment]
This issue is complete as of 5/22/2023 with commit `a1487fd`.  The main difficulty in moving a latest kernel & uboot was really the uboot version. Previous version relied on the uEnv.txt script for booting and i found as of 2021.10 release they moved to distro boot which looks for a extlinux file or a boot.scr file. This took me a long time to get to this conclusion though (process described in comments of issue [Add/Enable HX711 Driver](https://github.com/cu-ecen-aeld/final-project-geoffreyjensen6/issues/2). Once i got uboot updated though, the kernel update was very simple by just making a couple changes in menuconfig for the Kernel->Kernel Version = Latest Version (5.4) and also changing Toolchain->Kernel Headers = Linux 5.4.x kernel headers.  I did that and performed clean and build and it was able to boot. I did have to go back into make linux-menuconfig and reenable the HX711 driver again though.
